src/main/kotlin/com/hibob/rest_API/petController.kt

```diff
@@ -0,0 +1,79 @@
+package com.hibob.rest_API
+
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+import java.util.*
+
+val pets = mutableListOf(
+    Pet(1L, "Buddy", "Dog", "101", Date()),
+    Pet(2L, "Whiskers", "Cat", "102", Date()),
+    Pet(3L, "Polly", "Parrot", "103", Date())
+)
+
+
+
+@Controller
+@Path("/api/matan/pets/envelopes")
+@Produces(MediaType.APPLICATION_JSON)
+class PetController {
+
+    @GET
+    @Path("/getPetType/{petId}")
+    fun getPetType(@PathParam("petId") petId: Long): Response {
+        val pet = pets.find { it.petId == petId }
+
+        return when {
+            pet == null -> {
+                throw NotFoundException("Pet not found")  // Pet not found
+            }
+            petId == 123L -> {
+                Response.status(Response.Status.UNAUTHORIZED).build()  // Unauthorized for petId 123
+            }
+            else -> {
+                Response.ok(pet.type).build()
+            }
+        }
+    }
+
+    @GET
+    @Path("/getAllPets")
+    fun getAllPets(): Response {
+        return Response.ok(pets).build()
+    }
+
+    @POST
+    @Path("/addPet/{newPet}")
```

Comment:
Same here, no need addPet. Also this param in the url isn't used. The newPet is part of the payload/body of the request and not part of the url.